# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

<div class="span5 alert alert-info">
### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet
</div>
****

In [3]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [4]:
data = pd.io.stata.read_stata('C:\\Users\\miche\\Documents\\Springboard\\Racial_Disc_Project\\racial_disc\\data\\us_job_market_discrimination.dta')

In [5]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [6]:
sum(data[data.race=='w'].call)

235.0

In [7]:
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific    ...      compreq  orgreq  manuf  transcom  bankreal trade  \
0             17    ...          1.0     0.0    1.0       0.0       0.0   0.0   
1            316    ...          1.0     0.0    1.0       0.0       0.0   0.0   
2             19    ...          1.0     0.0    1.0       0.0       0.0   0.0   
3            313    ...          1.0     0.0    1.0       0.0       0.0   0.0   
4            313    ...          1.0     1.0    0.0       0.0       0.0   0.0   

  busservice othservice  missind  ownership  
0        0.0        0.0      0.0             
1        0.0        0.0      0.0             
2        0.0        0.0      0.0             
3        0.0        0.0      0.0             
4        0.0        1.0      0.0  Nonprofit  

[5 rows x 65 columns]

In [8]:
data.shape

(4870, 65)

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4870 entries, 0 to 4869
Data columns (total 65 columns):
id                    4870 non-null object
ad                    4870 non-null object
education             4870 non-null int8
ofjobs                4870 non-null int8
yearsexp              4870 non-null int8
honors                4870 non-null int8
volunteer             4870 non-null int8
military              4870 non-null int8
empholes              4870 non-null int8
occupspecific         4870 non-null int16
occupbroad            4870 non-null int8
workinschool          4870 non-null int8
email                 4870 non-null int8
computerskills        4870 non-null int8
specialskills         4870 non-null int8
firstname             4870 non-null object
sex                   4870 non-null object
race                  4870 non-null object
h                     4870 non-null float32
l                     4870 non-null float32
call                  4870 non-null float32
city        

In [10]:
data['race'].value_counts()

b    2435
w    2435
Name: race, dtype: int64

### 1. What test is appropriate for this problem? Does CLT apply?
The two-sample t test is appropriate for this test because we would like to the difference in the mean callback success rate for white-sounding names and the mean callback success rate for blck-sounding names is statistically significant. 

The CLT applies here because the sample size is large. When the sample size is large, the sampling distribution of the sample mean is approximately normal.

### 2. What are the null and alternate hypotheses?
The null hypothesis is that there is no difference in the mean callback success rate for white-sounding names and the mean callback success rate for black-sounding names.

The alternate hypothesis is that the difference between the mean callback success rate for white-sounding names and the mean callback success rate for black-sounding names does not equal to zero.

### 3. Compute margin of error, confidence interval, and p-value.

In [11]:
b=data[data.race=='b'].call
w=data[data.race=='w'].call
mean_diff=b.mean()- w.mean()
print("Mean callback success rate for black-sounding names:", b.mean())
print("Mean callback success rate for white-sounding names:", w.mean())
print("Difference in mean callback success rate:", mean_diff)

Mean callback success rate for black-sounding names: 0.0644763857126236
Mean callback success rate for white-sounding names: 0.09650924056768417
Difference in mean callback success rate: -0.03203285485506058


In [16]:
#Calculate margine of error and confidence interval at alpha =0.05
t=stats.t.ppf(0.975, len(b))
m=t*np.sqrt(b.var()/len(b)+w.var()/len(w))
ci=np.sort([mean_diff-m, mean_diff+m])
print('margin of error =', m)
print('95% confidence interval of difference in mean callback rate=',ci)

margin of error = 0.0152657250684
95% confidence interval of difference in mean callback rate= [-0.04729858 -0.01676713]


In [12]:
#Calculate t statistics and p-value
stats.ttest_ind(b, w) 

Ttest_indResult(statistic=-4.1147052908617514, pvalue=3.9408021031288859e-05)

### 4. Write a story describing the statistical significance in the context or the original problem.
The mean difference in callback success rate is -3.2%, meaning that overall the callback success rate for the group with black-sounding names is about 3.2% less that that with white-sounding names. 

The assumed 0 difference in mean is outside of the 95% confidence interval. The small p-value leads to the rejection of the null hypothesis that the race/name does not affect callback success rate and therefore the conclustion that the race/name affects the callback rate.

### 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?
Being statistically significant does not mean importance. The fact that it is statistically significant that the race/name factor affects the callback success rate does not mean it is the most important factor in determining the callback success rates. It just means that there is an impact but does not measure the magnitude of the impact. 

There are other factors that can affect the callback success rate such as education, skills, age, gender, etc. If I were to do the test again, I would look at records that only diff in race/name with all the factors being the same if it is possible to obtain such a dataset.